### PR TITLE
Example: Predition Intervals using Quantile Regression

### PR DESCRIPTION
**Pull request by billylanchantin**
## Description

This PR adds an example livebook inspired by the one from scikit-learn:

* https://scikit-learn.org/stable/auto_examples/ensemble/plot_gradient_boosting_quantile.html

It came out of a discussion on this issue:

* https://github.com/acalejos/exgboost/issues/5

## Notes

The livebook started as a fairly faithful adaption, but I ended up deviating in a number of ways:

* I use a normal instead of a log-normal distribution to build the noise. I started to make my own log-normal implementation (`Nx` doesn't have one), but I found that it was too distracting.
* I dropped the `MSE`-trained predictor. I found that it was too distracting as well since it was only used for comparison.
* I put more emphasis on the coverage and width metrics. I also changed the hyper-parameter tuning to tune for both at the same time.
* I added discussion sections at the end. They're not necessary, but it made the livebook feel more self-contained to me.

The section _Discussion: Pinball error_ felt the most ad hoc. But since the original example relied so heavily on [`mean_pinball_loss`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_pinball_loss.html#sklearn.metrics.mean_pinball_loss), I felt like I should include it.

It was also a way to bring up the early stopping issue: I couldn't tell what metric `EXGBoost` was using for early stopping. It should use the pinball error, but I think it may have been using `rmse` since I couldn't set `disable_default_eval_metric: false`. Though again, I'm not sure.

## Remarks

This it a fairly long example. If y'all think it's too long, I can cut it down.

Happy to take any other feedback as well!

### CODE REVIEW

**Comment by acalejos** [comment]
Hey, thanks for the PR! I'll take a look tonight

**Comment by acalejos** [comment]
Just took a look and ran it in Livebook and this is an awesome example @billylanchantin! Thanks so much for the contribution. Do you have a Twitter I can shout out in a tweet about this example?

**Comment by billylanchantin** [comment]
@acalejos Glad you liked it! I'm a big fan of worked examples so I'm happy I could add one.

No sorry I'm not on Twitter, but the offer is very kind :)
